activesupport/lib/active_support/message_pack/extensions.rb

```diff
@@ -0,0 +1,277 @@
+require "active_support/time"
+
+module ActiveSupport
+  module MessagePack
+    module Extensions # :nodoc:
+      extend self
+
+      def configure_factory(factory)
+        factory.register_type 0, Symbol,
+          packer: :to_msgpack_ext,
+          unpacker: :from_msgpack_ext,
+          optimized_symbols_parsing: true
+
+        factory.register_type 1, Integer,
+          packer: ::MessagePack::Bigint.method(:to_msgpack_ext),
+          unpacker: ::MessagePack::Bigint.method(:from_msgpack_ext),
+          oversized_integer_extension: true
+
+        factory.register_type 2, BigDecimal,
+          packer: :_dump,
+          unpacker: :_load
+
+        factory.register_type 3, Rational,
+          packer: method(:write_rational),
+          unpacker: method(:read_rational),
+          recursive: true
+
+        factory.register_type 4, Complex,
+          packer: method(:write_complex),
+          unpacker: method(:read_complex),
+          recursive: true
+
+        factory.register_type 5, Range,
+          packer: method(:write_range),
+          unpacker: method(:read_range),
+          recursive: true
+
+        factory.register_type 6, ActiveSupport::HashWithIndifferentAccess,
+          packer: method(:write_hash_with_indifferent_access),
+          unpacker: method(:read_hash_with_indifferent_access),
+          recursive: true
+
+        factory.register_type 7, Set,
+          packer: method(:write_set),
+          unpacker: method(:read_set),
+          recursive: true
+
+        factory.register_type 8, Time,
+          packer: method(:write_time),
+          unpacker: method(:read_time),
+          recursive: true
+
+        factory.register_type 9, DateTime,
+          packer: method(:write_datetime),
+          unpacker: method(:read_datetime),
+          recursive: true
+
+        factory.register_type 10, Date,
+          packer: method(:write_date),
+          unpacker: method(:read_date),
+          recursive: true
+
+        factory.register_type 11, ActiveSupport::TimeWithZone,
+          packer: method(:write_time_with_zone),
+          unpacker: method(:read_time_with_zone),
+          recursive: true
+
+        factory.register_type 12, ActiveSupport::TimeZone,
+          packer: method(:dump_time_zone),
+          unpacker: method(:load_time_zone)
+
+        factory.register_type 13, ActiveSupport::Duration,
+          packer: method(:write_duration),
+          unpacker: method(:read_duration),
+          recursive: true
+
+        factory.register_type 14, URI::Generic,
+          packer: :to_s,
+          unpacker: URI.method(:parse)
+
+        factory.register_type 15, IPAddr,
+          packer: :to_s,
+          unpacker: :new
+
+        factory.register_type 16, Pathname,
+          packer: :to_s,
+          unpacker: :new
+
+        factory.register_type 17, Regexp,
+          packer: :to_s,
+          unpacker: :new
+
+        factory.register_type 18, Module,
```

Comment:
I mainly added this as a "helper" for `write_object` / `read_object`.  In the context of a general-purpose serializer, do you think `write_object` / `read_object` should try to serialize classes for subsequent `from_msgpack_ext` or `json_create` calls?

We could possibly handle a missing class by returning a placeholder object rather than raising an error.

Reply:
> In the context of a general-purpose serializer

I think we'd need to define a general purpose serializer better (if that even exist). There's some very hard tradeoffs to make.

> We could possibly handle a missing class by returning a placeholder object rather than raising an error.

That would be very likely to just fail somewhere else where it will be much harder to debug.

Reply:
> That would be very likely to just fail somewhere else where it will be much harder to debug.

Quite possibly, though I was thinking of a hypothetical case where you want to remove a class from the code base but still gracefully handle old messages which reference that class.  `AS::MessagePack` could return a placeholder object that implements `#class_name` and `#data`, and the calling code could check for that placeholder and handle it accordingly.

Such behavior could be opt-in via a list of class names, and the default behavior could be to raise an error.